plugins/outputs/mqtt/mqtt.go

```diff
@@ -142,58 +145,67 @@ func (m *MQTT) Write(metrics []telegraf.Metric) error {
 }
 
 func (m *MQTT) publish(topic string, body []byte) error {
-	token := m.client.Publish(topic, byte(m.QoS), m.Retain, body)
-	token.WaitTimeout(time.Duration(m.Timeout))
-	if token.Error() != nil {
-		return token.Error()
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(m.Timeout))
+	defer cancel()
+	_, err := m.client.Publish(ctx, &paho.Publish{
```

Comment:
I guess the returned publish-response does not contain any error message or similar in case `err` is `nil`!?